### PR TITLE
Sync Updates + User Deletion

### PR DESCRIPTION
**Pull request by MatkoTom**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant
motivation and context. List any dependencies that are required for this change.

Implemented changes to the sync process. Everything is now paged. Also, added logic to delete user data from the app. Confirming deletion will remove all localDb, shared prefs and directory data.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can
reproduce. Please also list any relevant details for your test configuration. All testing was done
manually by following, plainly, functionality requirements. If there are some special edge cases or
special ways that things need to be tested they will be mentioned below.

**Test Configuration**:

* Firmware version: G980FXXSFFVHA
* Hardware: SAMSUNG Galaxy S20
* SDK: Android 13

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by ivolarevic — DISMISSED** [review]
LGTM

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -222,4 +222,5 @@
     <string name="failed_to_delete_block_for_user">Failed to delete block for user</string>
     <string name="failed_delete_note">Failed to delete note, please try again.</string>
     <string name="unknown_user">Unknown user</string>
+    <string name="dat_deletion_warning">Your user and all his data will be deleted</string>
```
Comment (ivolarevic) [inline app/src/main/res/values/strings.xml:225]: maybe you can rename this to data_deletion_warning

File: app/src/main/res/layout/fragment_settings.xml [hunk 2]
```diff
@@ -233,6 +233,29 @@
             app:layout_constraintTop_toTopOf="parent" />
     </androidx.constraintlayout.widget.ConstraintLayout>
 
+    <androidx.constraintlayout.widget.ConstraintLayout
```
Comment (ivolarevic) [inline app/src/main/res/layout/fragment_settings.xml:236]: can you maybe change this to frame layout or only use textview?
